stempy/image.cpp

```diff
@@ -186,5 +249,9 @@ Image<double> calculateAverage(std::vector<Block> &blocks)
   return image;
 }
 
-
+// Instantiate the ones that can be used
+template STEMImage createSTEMImage(StreamReader::iterator first,
+                                   StreamReader::iterator last, int rows,
+                                   int columns, int innerRadius,
+                                   int outerRadius);
```

Comment:
Not sure I understand what is going on here?

Reply:
Since we want to use iterators for `createSTEMImage`, it means that we need to make it a template function. Unfortunately, because it is a template function, it complicates the implementation. We have two choices:

1. Expose all of the implementation of `createSTEMImage` in the header file. This would currently also mean exposing some of the VTKm parts as well (although we could also make another function and move them to it), and including `ThreadPool` and many other files in the header file. A lot of implementation would be included and exposed in the `image.h` header file, and I wasn't sure if that's what we wanted.

2. Explicitly instantiate the template for all of the types we are allowing. This is what is happening above, and is the option I chose, just because it is simpler for now. It also effectively whitelists the function so that only the types we instantiate can be used. Right now, the template type `InputIt` can only be `StreamReader::iterator`, and possibly other types if we add them to the list. If we want to allow any user-defined type to be used with `createSTEMImage`, then we'll need to go with option number 1.

Reply:
Ok, lets add `std::vector<Block>::iterator`?

Reply:
Okay!